### PR TITLE
fix(slider): focus respective thumb when click on track for range slider

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #3109 

## Summary

This fix make sure the corresponding thumb is focused on` click `and the closest thumb `:focus` is set to true when the user click on track.

### CODE REVIEW

File: src/components/calcite-slider/calcite-slider.tsx [hunk 1]
```diff
@@ -1016,6 +1019,29 @@ export class CalciteSlider implements LabelableComponent {
...
   }
 
+  /**
+   * Get the closest thumb when clicked on the track
+   * @param event
+   * @returns {HTMLButtonElement}
+   * @internal
+   */
+  private getClosestHandle(event: PointerEvent): HTMLButtonElement {
+    return this.getDistanceX(this.maxHandle, event.clientX) >
```
[the remark this answers is not in the document]
Comment (anveshmekala) [reply src/components/calcite-slider/calcite-slider.tsx:1029]: Using the x value returned from `translate(event.clientX)` would work in most of the cases, except when we have `snap prop` as the values are rounded off.

File: src/components/calcite-slider/calcite-slider.tsx [hunk 2]
```diff
@@ -860,7 +861,7 @@ export class CalciteSlider implements LabelableComponent {
     const ticks = [];
     let current = this.min;
     while (this.ticks && current < this.max + this.ticks) {
-      ticks.push(current);
+      ticks.push(Math.min(current, this.max));
```
[the remark this answers is not in the document]
Comment (anveshmekala) [reply src/components/calcite-slider/calcite-slider.tsx:864]: This is fix from PR #3302 which is required for the tests.

File: src/components/calcite-slider/calcite-slider.scss [hunk 3]
```diff
@@ -308,6 +308,10 @@ $tick-height: 4px;
   background-color: var(--calcite-ui-brand);
 }
 
+.tick--active--range {
+  pointer-events: none;
```
[the remark this answers is not in the document]
Comment (anveshmekala) [reply src/components/calcite-slider/calcite-slider.scss:312]: If set the `pointer-events:none` for non-range ticks then the users can't move the thumb to tick values . Only keyboard would help them navigate the thumb to tick values. Am i understanding it right?

[the remark this answers is not in the document]
Comment (anveshmekala) [reply src/components/calcite-slider/calcite-slider.scss:312]: Nevermind, the user's are still able to move the thumb to tick values and the events can bubble.

File: src/components/calcite-slider/calcite-slider.tsx [hunk 4]
```diff
@@ -547,7 +547,8 @@ export class CalciteSlider implements LabelableComponent {
                   <span
                     class={{
                       tick: true,
-                      "tick--active": activeTicks
+                      "tick--active": activeTicks,
+                      "tick--active--range": this.isRange
```
[the remark this answers is not in the document]
Comment (anveshmekala) [reply src/components/calcite-slider/calcite-slider.tsx:551]: I was thought `tick--range` was already taken 🥲.

File: src/components/calcite-slider/calcite-slider.e2e.ts [hunk 5]
```diff
@@ -178,6 +178,70 @@ describe("calcite-slider", () => {
...
+    await page.waitForChanges();
+
+    const isMaxThumbFocused = await page.$eval("calcite-slider", (slider) =>
+      slider.shadowRoot.activeElement?.classList.contains("thumb--value")
+    );
+
+    expect(await slider.getProperty("minValue")).toBe(0);
+    expect(await slider.getProperty("maxValue")).toBe(80);
+    expect(isMaxThumbFocused).toBe(true);
+  });
```
[the remark this answers is not in the document]
Comment (anveshmekala) [reply src/components/calcite-slider/calcite-slider.e2e.ts:243]: Yup. Added a new test for clicking in the middle focuses the `max-thumb`. Correct me if i am wrong @Esri/calcite-design-representatives .

File: src/components/calcite-slider/calcite-slider.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (anveshmekala) [reply src/components/calcite-slider/calcite-slider.tsx:865]: never mind . will revert this change.

**Review by jcfranco — APPROVED** [review]
Awesome stuff! 🎉 :taco::t-rex:

**Pull request merged**